nomad/state/state_store.go

```diff
@@ -1235,84 +1286,122 @@ func (s *StateStore) updateSummaryWithJob(job *structs.Job, txn *memdb.Txn) erro
 				Starting: 0,
 			}
 			existing.Summary[tg.Name] = newSummary
+			hasSummaryChanged = true
+		}
+	}
+
+	// The job summary has changed, so add to watcher and update the modify
+	// index.
+	if hasSummaryChanged {
+		existing.ModifyIndex = index
+		watcher.Add(watch.Item{Table: "job_summary"})
+		watcher.Add(watch.Item{JobSummary: job.ID})
+
+		// Update the indexes table for job summary
+		if err := txn.Insert("index", &IndexEntry{"job_summary", index}); err != nil {
+			return fmt.Errorf("index update failed: %v", err)
 		}
 	}
 
-	if err := txn.Insert("job_summary", existing); err != nil {
+	if err := txn.Insert("job_summary", *existing); err != nil {
```

Comment:
Doesn't matter since the object won't change. But doesn't harm to do this inside the block.